gateway/sensor.go

```diff
@@ -92,13 +92,23 @@ func newGateway(
 	conf resource.Config,
 	logger logging.Logger,
 ) (sensor.Sensor, error) {
+	cfg, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewGateway(ctx, conf.ResourceName().AsNamed(), cfg, logger)
+}
+
```

Comment:
Are you exporting for your cmd script? We don't usually do this in our modules.

Reply:
huh? why not... it's insanely useful

Reply:
Then, you can just export the `newGateWay` constructor function so we're using the same resource interfaces and conf for both. I'd rather do that since `ResourceName().AsNamed()` gives me the rdk's DoCommand as well.